lms/djangoapps/shoppingcart/processors/MPesa.py

```diff
@@ -0,0 +1,77 @@
+import json
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from edxmako.shortcuts import render_to_string
+from shoppingcart.processors.helpers import get_processor_config
+from shoppingcart.models import Order
+from mpesa_api.tasks import check_paymet
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def render_purchase_form_html(cart, callback_url=''):
+    return render_to_string('shoppingcart/mpesa_form.html', {
+        'action': get_processor_config().get('PURCHASE_ENDPOINT'),
+        'params': {
+            'cmd': '_xclick',
+            'charset': 'utf-8',
+            'currency_code': cart.currency.upper(),
+            'amount': cart.total_cost,
+            'item_name': "Camara:{}".format(cart.total_cost),
+            'custom': cart.id,
+            'business': get_processor_config().get('CLIENT_ID'),
+            'notify_url': callback_url,
+            'cancel_return': 'https://{}{}'.format(settings.SITE_NAME, reverse('shoppingcart.views.show_cart')),
+            'return': 'https://{}{}'.format(settings.SITE_NAME, reverse('dashboard')),
+        }
+    })
+
+
+def process_postpay_callback(params):
+    mpesa_result = check_paymet(int(params['order_id']))
+    order = Order.objects.get(id=int(mpesa_result['custom']))
+    if mpesa_result['payment_status'] == 'ok':
+        log.info('Order "{}" and transaction "{}" is successed'.format(order, mpesa_result['txn_id']))
+        order.purchase(
+            processor_reply_dump=json.dumps(params)
+        )
+        return {'success': True, 'order': order, 'error_html': ''}
+    else:
+        log.error('Order "{}" and transaction "{}" is failed'.format(order, mpesa_result['txn_id']))
+        return {'success': False, 'order': order, 'error_html': 'Transaction "{}" is filed'.format(mpesa_result['txn_id'])}
+
+def get_purchase_endpoint():
+    """
+    Return the URL of the payment end-point for CyberSource.
+
+    Returns:
+        unicode
+
```

Comment:
An odd empty line could be omitted.

Reply:
Done (https://github.com/raccoongang/edx-platform/pull/536)